src/app/shared/editor/components/yaml-editor/yaml-editor.component.ts

```diff
@@ -0,0 +1,43 @@
+import {
+    ChangeDetectionStrategy,
+    Component,
+    EventEmitter,
+    Input,
+    OnChanges,
+    Output,
+    SimpleChanges,
+} from "@angular/core";
+
+import * as monaco from "monaco-editor";
+import { MaybeNull } from "../../../../common/app.types";
+import { MonacoService } from "../../services/monaco.service";
+import { YAML_EDITOR_OPTIONS } from "./config-editor.events";
+
+@Component({
+    selector: "app-yaml-editor",
+    templateUrl: "./yaml-editor.component.html",
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class YamlEditorComponent implements OnChanges {
+    @Input() public yamlTemplate = "";
+    @Input() public yamlError: MaybeNull<string>;
+    @Input() public readonly YAML_EDITOR_OPTIONS: monaco.editor.IStandaloneEditorConstructionOptions =
+        YAML_EDITOR_OPTIONS;
```

Comment:
Why is this an input? It does not contain any data that look like I would want to program from outside.

Reply:
I think it's a good way to handle options for third-party components, to have the ability to pass options as input.
But I can make them pre-defined, it's not hard to male them input in the future

Reply:
Changed